src/clientHandler.ts

```diff
@@ -12,298 +10,177 @@ import {
+    const executable: Executable = {
+      command: cmd,
+      args: serverArgs,
+      options: {},
+    };
+    const serverOptions: ServerOptions = {
+      run: executable,
+      debug: executable,
+    };
+    return serverOptions;
+  }
 
-    if (this.supportsMultiFolders) {
-      promises.push(this.stopClient(MULTI_FOLDER_CLIENT));
-      return Promise.all(promises);
-    }
+  private getInitializationOptions(commandPrefix: string) {
+    const rootModulePaths: string[] = config('terraform-ls').get('rootModules');
+    const terraformExecPath: string = config('terraform-ls').get('terraformExecPath');
+    const terraformExecTimeout: string = config('terraform-ls').get('terraformExecTimeout');
+    const terraformLogFilePath: string = config('terraform-ls').get('terraformLogFilePath');
+    const excludeModulePaths: string[] = config('terraform-ls').get('excludeRootModules');
+    const ignoreDirectoryNames: string[] = config('terraform-ls').get('ignoreDirectoryNames');
 
-    if (!folders) {
-      folders = [];
-      for (const key of clients.keys()) {
-        folders.push(key);
-      }
+    if (rootModulePaths.length > 0 && excludeModulePaths.length > 0) {
+      throw new Error(
+        'Only one of rootModules and excludeRootModules can be set at the same time, please remove the conflicting config and reload',
+      );
     }
 
-    for (const folder of folders) {
-      promises.push(this.stopClient(folder));
+    const experimentalFeatures = config('terraform-ls').get('experimentalFeatures');
+    let initializationOptions = { commandPrefix, experimentalFeatures };
+    if (terraformExecPath.length > 0) {
+      initializationOptions = Object.assign(initializationOptions, { terraformExecPath });
+    }
+    if (terraformExecTimeout.length > 0) {
+      initializationOptions = Object.assign(initializationOptions, { terraformExecTimeout });
     }
-    return Promise.all(promises);
+    if (terraformLogFilePath.length > 0) {
+      initializationOptions = Object.assign(initializationOptions, { terraformLogFilePath });
+    }
+    if (rootModulePaths.length > 0) {
+      initializationOptions = Object.assign(initializationOptions, { rootModulePaths });
+    }
+    if (excludeModulePaths.length > 0) {
+      initializationOptions = Object.assign(initializationOptions, { excludeModulePaths });
+    }
+    if (ignoreDirectoryNames.length > 0) {
+      initializationOptions = Object.assign(initializationOptions, { ignoreDirectoryNames });
+    }
+    return initializationOptions;
   }
 
-  private async stopClient(folder: string): Promise<void> {
-    if (!clients.has(folder)) {
-      console.log(`Attempted to stop a client for folder: ${folder} but no client exists`);
-      return;
+  public async stopClient(): Promise<void> {
+    if (this.tfClient === undefined) {
+      return Promise.resolve();
+    }
+
+    if (this.tfClient.client === undefined) {
```

Comment:
I think we can ditch the first if when we use optional chaining here.
```suggestion
    if (this.tfClient?.client === undefined) {
```

and just return without the `Promise.resolve()`